numpy/core/src/multiarray/nditer_pywrap.c

```diff
@@ -2594,6 +2562,25 @@ npyiter_discard_writeback(NewNpyArrayIterObject *self)
     Py_RETURN_NONE;
 }
 
+static PyObject *
+npyiter_exit(NewNpyArrayIterObject *self, PyObject *args)
+{
+    PyObject *exc_type, *exc_value, *traceback;
+    if (!PyArg_ParseTuple(args, "OOO", &exc_type, &exc_value, &traceback)) {
+        Py_RETURN_NONE;
+    }
+    if (self->iter == NULL) {
+        Py_RETURN_NONE;
+    }
+    if (exc_type == Py_None) {
+        npyiter_resolve_writeback(self);
+    } else {
+        npyiter_discard_writeback(self);
```

Comment:
I think @ahaldane is suggesting these become part of the public interface to NpyIter, not just to `np.nditer`. Am I right in saying that?

Reply:
IMO that's a good idea *except* for the `npyiter_discard_writeback`, because we're considering removing that from Python as well.